bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,819 @@
+    @SuppressWarnings({ "deprecation", "null" })
+    public void closeWebSocketConnection() {
+        if (socketMonitor != null) {
+            socketMonitor.stop();
+        }
+
+        if (jettyThreadPool != null) {
+            try {
+                jettyThreadPool.stop();
+            } catch (Exception e1) {
+                logger.debug("Error by closing Websocket connection [{}]", e1.getMessage());
+            }
+        }
+
+        if (websocketClient != null) {
+            try {
+                websocketClient.stop();
+            } catch (Exception e2) {
+                logger.debug("Error by closing Websocket connection [{}]", e2.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Method to open the websocket connection
+     *
+     * @author Andras Uhrin
+     *
+     */
+    @SuppressWarnings("null")
+    public boolean openWebSocketConnection() {
+        boolean ret = true;
+
+        String authString = username + ":" + password;
+
+        // create base64 encoder
+        Base64.Encoder bas64Encoder = Base64.getEncoder();
+
+        // Encoding string using encoder object
+        String authStringEnc = bas64Encoder.encodeToString(authString.getBytes());
+
+        authField = "Basic " + authStringEnc;
+
+        if (jettyThreadPool == null) {
+            jettyThreadPool = new QueuedThreadPool();
+            jettyThreadPool.setName(FreeAtHomeBridgeHandler.class.getSimpleName());
+            jettyThreadPool.setDaemon(true);
+            jettyThreadPool.setStopTimeout(0);
+        }
+
+        if (websocketClient == null) {
+            websocketClient = new WebSocketClient(httpClient);
+
+            if (websocketClient != null) {
+                websocketClient.setExecutor(jettyThreadPool);
+            }
+        }
+
+        if (socket == null) {
+            socket = new EventSocket();
+
+            if (socket != null) {
+                // set bridge for the socket event handler
+                socket.setBridge(this);
+
+                if (socketMonitor == null) {
+                    socketMonitor = new FreeAtHomeWebsocketMonitorThread();
+                    socketMonitor.start();
+
+                    ret = true;
+                } else {
+                    websocketClient = null;
+                    socket = null;
+                    jettyThreadPool = null;
+                    ret = false;
+
+                    logger.debug("Cannot open http connection - socketmonitor");
+                }
+            }
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to re-initialize the bridge
+     *
+     * @author Andras Uhrin
+     *
+     */
+    @Override
+    public void thingUpdated(Thing thing) {
+        dispose();
+        this.thing = thing;
+        initialize();
+    }
+
+    /**
+     * Method to initialize the bridge
+     *
+     * @author Andras Uhrin
+     *
+     */
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
This can be postponed until before starting async task. Otherwise you risk two status updates, for example if configuration is invalid.

Reply:
Statement removed. It is not needed

Reply:
I think you will need it, see https://github.com/openhab/openhab-addons/pull/13703/files#r1166007325

Reply:
@andrasU did you see this somewhat buried comment?

Reply:
Sorry, you are still making two status updates, but now they are both async, so you risk leaving `initialize()` without any status update. I will make a more clear proposal towards the new code.

Reply:
It should be moved down so that status **UNKNOWN** is set after validating the configuration and before scheduling the Runnable. Otherwise you'll see a quick transition from **INITIALIZING** → **UNKNOWN** → **OFFLINE** on configuration errors.

Reply:
**Fixed**

Thanks the clarity with your description